doc-design/source/softwareArchitecture.rst

```diff
@@ -1157,9 +1158,9 @@ QSS performance considerations:
 
 - Avoiding global algebraic solutions when doing atomic/subset lookups when the algebraic dependencies
   can be partitioned is probably difficult but worth investigating.
+  [Stuart: Can you elaborate? I don't quite
+  understand. Should this be in a section "Getting individual outputs"? Is a "lookup" a call to fmi2GetReal?]
```

Comment:
Maybe a section called "Atomic Operations" would be a good name. This is all about making sure that atomic (individual variable/derivative get/set operations) trigger the minimal amount of work in the FMU internals. This means that an atomic getReal (for a state variable or its derivative) should not refresh all derivatives and if an algebraic solution is necessary that, ideally, it is as "local"/minimal as possible. I assume that, given traditional solvers, this has not been a priority in the past so we will want Modelon to look at the entire FMI API and any new extensions for places where unnecessary global computations occur. Before this becomes a requirement we should discuss with Modelon to better understand the current state of the art in OCT code generation. Since I can't generate source from OCT with the current version I can't investigate myself other than by doing performance profiling.